### ISSUE TITLE
meaning of last_updated field for projects

### ISSUE THREAD

**Issue by cjerdonek** [codeforamerica/cfapi#219]
I've noticed over the past month or so that on Code for SF's [projects page](http://codeforsanfrancisco.org/projects/), which is ordered by "last updated," one project always seems to show up as having been updated within the last hour or so (at all hours of the week).  When I started looking into the source of the project's amazing productivity, I noticed that the project often doesn't seem to have any recent commits.  It makes me wonder about the meaning of the `last_updated` field.  I'm speculating here, but I wonder if enabling certain GitHub services can cause a project to show up as updated on a regular basis (e.g. automated test runners, etc).  Thoughts?

**Comment by ondrae** [comment]
Hey @cjerdonek you've found another bug, one that I thought we already took care of. That Ballot API project seems to be falsely getting updated every hour, which is how often the CfAPI updates. We're checking on it today.

To answer your questions though, `last_updated` for GitHub based projects comes from the GitHub `updated_at` field. This includes not just commits and file updates, but also stars, watchers, and issues. We've debated making it only from actual commits.

We'll let you know when we fix this, so that your Election Data project shows back up at the top.

**Comment by ondrae** [comment]
Hey @tmaybe when Code-for-San-Francisco updates, here are the logs:
```
INFO:root:Asking for https://docs.google.com/spreadsheet/pub?key=0ArHmv-6U1drqdDVGZzdiMVlkMnRJLXp2cm1ZTUhMOFE&output=csv
INFO:root:Asking Github for https://api.github.com/repos/sfbrigade/ballotapi ({'If-Modified-Since': 'Thu, 09 Apr 2015 19:05:43 GMT'})
INFO:root:Project https://api.github.com/repos/sfbrigade/ballotapi has not been modified since last update
INFO:root:Asking Github for https://api.github.com/repos/sfbrigade/ballotapi/contents/ ({'If-None-Match': u'W/"ce172e7b661858e4afbb67f8549d1346"'})
INFO:root:root directory listing has not changed since last update for https://api.github.com/repos/sfbrigade/ballotapi/contents/
INFO:root:Asking Github for https://api.github.com/repos/sfbrigade/ballotapi ({'If-Modified-Since': 'Thu, 09 Apr 2015 19:05:43 GMT'})
INFO:root:Project https://api.github.com/repos/sfbrigade/ballotapi has not been modified since last update
INFO:root:Asking Github for https://api.github.com/repos/sfbrigade/ballotapi/contents/ ({'If-None-Match': u'W/"ce172e7b661858e4afbb67f8549d1346"'})
INFO:root:root directory listing has not changed since last update for https://api.github.com/repos/sfbrigade/ballotapi/contents/
INFO:root:Project https://api.github.com/repos/sfbrigade/ballotapi has been modified via spreadsheet or civic.json.
```

Something is happening when it gets read in by the spreadsheet.

**Comment by cjerdonek** [comment]
Cool, thanks! :wink:

**Comment by ondrae** [comment]
I found it. Code for San Francisco uses [a spreadsheet](https://docs.google.com/a/codeforamerica.org/spreadsheet/ccc?key=0ArHmv-6U1drqdDVGZzdiMVlkMnRJLXp2cm1ZTUhMOFE#gid=0) for their projects list. Ballot API was listed twice in there, with two different descriptions. So every hour the first description would be different than what we had in our database and get updated, just to get rewritten again by the second description.

I deleted the duplicate entry. Should show up in the next hourly update.
